frontend/src/app/components/wp-table/timeline/container/wp-timeline-container.directive.ts

```diff
@@ -416,4 +471,77 @@ export class WorkPackageTimelineTableController extends UntilDestroyedMixin impl
       }
     }
   }
+
+  manageCollapsedGroupHeaderCells(allGroups:GroupObject[],
+                                  collapsedGroupsChange:{[key:string]:boolean}|undefined,
+                                  tableWorkPackages:WorkPackageResource[],
+                                  collapsedGroupsCellsMap:IGroupCellsMap,
+                                  updateAllHeaderCells?:boolean) {
+    if (!collapsedGroupsChange) {
+      return;
+    }
+
+    const collapsedGroupChangesToManage = Object.keys(collapsedGroupsChange).filter(groupKey => {
+      const keyGroupType = groupKey.split('-')[0];
+
+      return this.groupTypesWithHeaderCellsWhenCollapsed.includes(keyGroupType);
+    });
+    let groupsToUpdate:string[];
+
+    if (updateAllHeaderCells) {
+      groupsToUpdate = collapsedGroupChangesToManage;
+    } else {
+      groupsToUpdate = collapsedGroupChangesToManage.filter(groupKey => {
+        const currentGroupCollapsedValue = collapsedGroupsChange[groupKey];
+        const storedGroup = allGroups.find(group => group.identifier === groupKey);
+
+        return storedGroup && storedGroup.collapsed !== currentGroupCollapsedValue;
+      });
+    }
```

Comment:
It seems to me that all of this is still filtering and checking wether any changes are actually necessary to be made. Because of that it would make sense to place it as some `filter` in the `pipe` that is calling `manageCollapsedGroupHeaderCells`